app/contracts/work_packages/create_contract.rb

```diff
@@ -37,15 +37,21 @@ class CreateContract < BaseContract
 
     default_attribute_permission :add_work_packages
 
-    validate :user_allowed_to_add
+    validate :user_allowed_to_add_work_packages
+    validate :user_allowed_to_manage_file_links
 
     private
 
-    def user_allowed_to_add
+    def user_allowed_to_add_work_packages
       if (model.project && !@user.allowed_to?(:add_work_packages, model.project)) ||
          !@user.allowed_to_globally?(:add_work_packages)
+        errors.add(:base, :error_unauthorized)
+      end
+    end
 
-        errors.add :base, :error_unauthorized
+    def user_allowed_to_manage_file_links
+      if model.file_links.present? && model.project.present? && !user.allowed_to?(:manage_file_links, model.project)
```

Comment:
Here I am not sure if it is the right place for error to be raised. Actually we could raise an error while processing passed file_links(even do not parse them if the user is not allowed to manage file_links). Then it would save some resources(for example, finding file_links in the db and initializing models.
Is it allowed somehow by design of the API?